### PR TITLE
add user edit profile

### PR DESCRIPTION
**Pull request by annaguidi**
add user edit profile

### CODE REVIEW

File: app/controllers/profiles_controller.rb [hunk 1]
```diff
@@ -3,4 +3,29 @@ def show
...
+    end
+  end
+
+  private
+
+  def profile_params
+    params.require(:profile).permit(
+      :username,
+      :location,
+      :pun
```
Comment (alxjrvs) [inline app/controllers/profiles_controller.rb]: Hmm.

File: spec/features/user_edits_their_profile_spec.rb [hunk 2]
```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+# [] Visiting the `/landmarks/1` path should show the landmark details for a landmark with the ID of 1.
+
+feature "user edits their profile" do
+  scenario "successfully edits profile" do
+    user = FactoryGirl.create(:user)
+
+    profile = Profile.create(
```
Comment (alxjrvs) [inline spec/features/user_edits_their_profile_spec.rb]: This is better off as a factory.

File: spec/features/user_edits_their_profile_spec.rb [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    click_button "Update Profile"
+
+    expect(page).to have_content "BillyBob"
+    expect(page).to have_content "France"
+    expect(page).to have_content "Know thy pun?"
+    expect(page).to have_content "Successfully edited your Profile!"
+  end
+
+  scenario "successfully edits profile" do
+    user = FactoryGirl.create(:user)
```
Comment (alxjrvs) [inline spec/features/user_edits_their_profile_spec.rb]: consider using `let` syntax here.

File: spec/features/user_edits_their_profile_spec.rb (hunk 2 above)
Comment (annaguidi) [inline spec/features/user_edits_their_profile_spec.rb]: we're having trouble relating these two classes if we use a factory for both, which is why we are manually passing in user for Profile

**Pull request merged**